src/components/member-role-update/index.js

```diff
@@ -26,48 +23,98 @@ const MemberRoleUpdate = () => {
     `${BASE_API_URL}/users/${selectedMember}`
   );
 
+  const member = userData?.user?.roles.member;
+  const archived = userData?.user?.roles.archived;
+  const userId = userData?.user.id;
   const { data: tagData } = useFetch(`${BASE_API_URL}/tags`);
   const { data: levelData } = useFetch(`${BASE_API_URL}/levels`);
 
-  const promoteToMember = async (user) => {
+  const promoteDemoteAMember = async (id) => {
+    let memberRole = null;
     setIsUpdating(true);
-    const { status } = await moveToMember(user);
-    setIsUpdating(false);
-    if (status === 204) {
-      setUpdateStatus('User moved to member');
+    if (member) {
+      memberRole = false;
     } else {
-      setUpdateStatus('Some error occured, please contact admin');
+      memberRole = true;
+    }
+    const role = {
+      member: memberRole,
+    };
+    try {
+      const { status } = await memberRoleUpdate(id, role);
+      setIsUpdating(false);
+      if (status === 200) {
+        setUpdateStatus('user promoted to a member');
+      }
+    } catch (error) {
+      setUpdateStatus('some error occured, please contact admin');
     }
   };
-  const archiveTheMember = async (user) => {
+
+  const archiveUnArchiveTheMember = async (id) => {
+    let archiveRole = null;
     setIsUpdating(true);
-    const { status } = await archiveMember(user);
-    setIsUpdating(false);
-    if (status === 204) {
-      setUpdateStatus('User archived!');
+    if (archived) {
+      archiveRole = false;
     } else {
+      archiveRole = true;
+    }
+    const role = {
+      archived: archiveRole,
+    };
+    try {
+      const { status } = await memberRoleUpdate(id, role);
+      setIsUpdating(false);
+      if (status === 200) {
+        setUpdateStatus('user archived!');
+      }
+    } catch (error) {
       setUpdateStatus('Some error occured, please contact admin');
     }
   };
 
+  const memberRoleUpdateButton = member ? (
+    <button
+      className={classNames.moveToMember}
+      type="button"
+      onClick={() => promoteDemoteAMember(userId)}
+    >
+      Demote Member
+    </button>
+  ) : (
+    <button
+      className={classNames.moveToMember}
+      type="button"
+      onClick={() => promoteDemoteAMember(userId)}
+    >
+      Promote to Member
+    </button>
+  );
```

Comment:
https://github.com/Real-Dev-Squad/website-members/pull/536#discussion_r1275181855 I don't think that this comment is resolved yet, so you created a function but I wanted to have just one button for it and check what text to show and change the payload of the function as well by checking the status only